src/lib.rs

```diff
@@ -0,0 +1,339 @@
+use chrono::prelude::*;
+use log::{error, info, warn};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use sha2::{Digest, Sha256};
+
+const DIFFICULTY_PREFIX: &str = "00";
+
+pub struct App {
+    pub blocks: Vec<Block>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct Block {
+    pub id: u64,
+    pub hash: String,
+    pub previous_hash: String,
+    pub timestamp: i64,
+    pub data: String,
+    pub nonce: u64,
+}
+
+fn hash_to_binary_representation(hash: &[u8]) -> String {
+    let mut rep: String = String::default();
+    for c in hash {
+        rep.push_str(&format!("{:b}", c));
+    }
+    rep
+}
+
+fn calculate_hash(id: u64, timestamp: i64, previous_hash: &str, data: &str, nonce: u64) -> Vec<u8> {
+    let object = json!({
+        "id": id,
+        "previous_hash": previous_hash,
+        "data": data,
+        "timestamp": timestamp,
+        "nonce": nonce
+    });
+
+    let mut hasher = Sha256::new();
+    hasher.update(object.to_string().as_bytes());
+    hasher.finalize().as_slice().to_owned()
+}
+
+impl App {
+    pub fn default() -> Self {
```

Comment:
to powinna być implementacja traita Default

Reply:
clippy na to zwraca uwagę

Reply:
:+1: